src/main/java/com/ardaslegends/repository/war/battle/BattleRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.ardaslegends.repository.war.battle;
+
+import com.ardaslegends.domain.war.battle.Battle;
+import com.ardaslegends.domain.war.battle.QBattle;
+import com.ardaslegends.repository.exceptions.NotFoundException;
+import com.ardaslegends.repository.exceptions.RepositoryNullPointerException;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.data.jpa.repository.support.QuerydslRepositorySupport;
+
+@Slf4j
+public class BattleRepositoryImpl extends QuerydslRepositorySupport implements BattleRepositoryCustom {
+
+
+    public BattleRepositoryImpl() { super(Battle.class); }
+
+    @Override
+    public Battle findByIdOrElseThrow(Long id) {
```

Comment:
It was current convention to call queryDsl functions with an initial "query" instead of "find"

Reply:
Done